File: 0621-task-scheduler/0621-task-scheduler.java
class Solution {
    public int leastInterval(char[] tasks, int n) {
        int len = tasks.length;
        HashMap<Character, Integer> map = new HashMap<>();
        int maxFreq = 0;
        int noOfMaxFreq = 0;
        for (char c : tasks) {
            map.put(c, map.getOrDefault(c, 0) + 1);
            maxFreq = Math.max(maxFreq, map.get(c));
        }
        for (Map.Entry<Character, Integer> entry : map.entrySet()) {
            if (entry.getValue() == maxFreq) {
                noOfMaxFreq++;
            }
        }
        int res = Math.max((n + 1) * (maxFreq - 1) + noOfMaxFreq, len);
        return res;
    }
}

File: 0621-task-scheduler/0621-task-scheduler.cpp
class Solution {
public:
    int leastInterval(vector<char>& tasks, int n) {
        int len = tasks.size();
        unordered_map<char, int> map;
        int maxFreq = 0;
        int noOfMaxFreq = 0;
        for(char c: tasks) {
            map[c]++;
            maxFreq = max(maxFreq, map[c]);
        }
        for(pair<int, int> x: map)
            if(x.second == maxFreq)
                noOfMaxFreq++;
        int res = max(((n+1)*(maxFreq-1)+noOfMaxFreq), len);
        return res;
    }
};